plugins/source/gcp/resources/services/securitycenter/findings_fetch.go

```diff
@@ -2,28 +2,62 @@ package securitycenter
 
 import (
 	"context"
+	"fmt"
+	"time"
 
 	"google.golang.org/api/iterator"
 
 	pb "cloud.google.com/go/securitycenter/apiv1/securitycenterpb"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugins/source/gcp/client"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	securitycenter "cloud.google.com/go/securitycenter/apiv1"
 )
 
-func fetchFindings(parent string) func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-	return func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-		c := meta.(*client.Client)
-		req := &pb.ListFindingsRequest{
+const pageSize = 1000
+
+func getRequest(ctx context.Context, c *client.Client, table string, parent string) *pb.ListFindingsRequest {
+	filter, err := c.Backend.Get(ctx, table, c.ID())
+	if err != nil {
+		c.Logger().Warn().Str("table", table).Msgf("failed to get filter %s", err.Error())
+	}
+	var req *pb.ListFindingsRequest
+	if filter == "" || err != nil {
+		req = &pb.ListFindingsRequest{
```

Comment:
I would prepare a single `req` and then set `filter` if it's available. Also, since filter is a string and default zero value is already "" we don't need this if and "selective setting" at all.

Reply:
Good point I'll update it

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/7696/commits/88810fbf4e38c0df498fc02a3476410c904b0cdb